### PR TITLE
partition manager impl

### PR DESCRIPTION
**Pull request by dinal**

### CODE REVIEW

File: pkg/appender/store.go [hunk 1]
```diff
@@ -218,7 +221,8 @@ func (cs *chunkStore) chunkByTime(t int64) *attrAppender {
 		if err != nil {
 			return nil
 		}
-		cur.initialize(part.NextPart(t), t) // TODO: next part
+		nextPart, _  := part.NextPart(t)
```
Comment (imakhlin) [inline pkg/appender/store.go]: Why error is ignored here?

File: pkg/config/config.go [hunk 2]
```diff
@@ -52,42 +53,52 @@ type V3ioConfig struct {
...
+	Aggregators []string `json:"aggregators"`
+	AggregatorsGranularityInSeconds int `json:"aggregatorsGranularityInSeconds"`
+	StorageClass string `json:"storageClass"`
+	SampleRetention int `json:"sampleRetention"`
+	PartitionerInterval string `json:"partitionerInterval"`
+	ChunckerInterval string `json:"chunckerInterval"`
+}
+
+type Partition struct {
+	StartTime int64 `json:"startTime"`
```
Comment (imakhlin) [inline pkg/config/config.go]: Units! What time is expected here? I guess time in seconds but maybe I'm wrong.

File: pkg/partmgr/partmgr.go [hunk 3]
```diff
@@ -27,33 +27,47 @@ import (
...
-		hoursInChunk:  pmgr.cfg.HrInChunk,
+		path:          path,
+		startTime:     startTime,
+		partitionInterval: IntervalInMilli(pmgr.cfg.PartitionSchemaInfo.PartitionerInterval),
+		chunkInterval: IntervalInMilli(pmgr.cfg.PartitionSchemaInfo.ChunckerInterval),
 		prefix:        "",
-		retentionDays: pmgr.cfg.DaysRetention,
-		rollupTime:    int64(pmgr.cfg.RollupMin) * 60 * 1000,
+		retentionDays: pmgr.cfg.PartitionSchemaInfo.SampleRetention,
+		rollupTime:    int64(pmgr.cfg.PartitionSchemaInfo.AggregatorsGranularityInSeconds) * 60 * 1000,
```
Comment (imakhlin) [inline pkg/partmgr/partmgr.go]: Why don't we align everything to Millis or to Seconds?

File: pkg/partmgr/partmgr.go [hunk 4]
```diff
@@ -62,49 +76,96 @@ func NewDBPartition(pmgr *PartitionManager) *DBPartition {
...
 	return p.cfg
 }
 
 func (p *PartitionManager) Init() error {
-
 	return nil
 }
 
-func (p *PartitionManager) TimeToPart(t int64) *DBPartition {
+func (p *PartitionManager) TimeToPart(t int64) (*DBPartition, error) {
```
Comment (imakhlin) [inline pkg/partmgr/partmgr.go]: I guess `t` suppose to be `timeInMilliseconds`

File: pkg/partmgr/partmgr.go [hunk 5]
```diff
@@ -62,49 +76,96 @@ func NewDBPartition(pmgr *PartitionManager) *DBPartition {
...
 }
 
-func (p *PartitionManager) PartsForRange(mint, maxt int64) []*DBPartition {
+func (p *PartitionManager) updatePartitionInSchema(partition *DBPartition) error {
+	p.cfg.Partitions = append(p.cfg.Partitions, config.Partition{StartTime:partition.startTime, SchemaInfo: p.cfg.PartitionSchemaInfo})
+	data, err := json.Marshal(p.cfg)
+	if err != nil {
+		return errors.Wrap(err, "Failed to update new partition in schema file")
+	}
+	err = p.container.Sync.PutObject(&v3io.PutObjectInput{Path:p.path + config.SCHEMA_CONFIG, Body: data})
```
Comment (imakhlin) [inline pkg/partmgr/partmgr.go]: Naming convention, should be `SchemaConfig`

File: pkg/partmgr/partmgr.go [hunk 6]
```diff
@@ -62,49 +76,96 @@ func NewDBPartition(pmgr *PartitionManager) *DBPartition {
...
 	path           string             // Full path (in the DB) to the partition
-	partID         int                // PartitionID
 	startTime      int64              // Start from time/date
-	days           int                // Number of days stored in the partition
-	hoursInChunk   int                // number of hours stored in each chunk
+	partitionInterval int64             // Number of millis stored in the partition
+	chunkInterval  int64                // number of millis stored in each chunk
 	prefix         string             // Path prefix
-	retentionDays  int                // Keep samples for N days
+	retentionDays  int                // Keep samples for N hours
```
Comment (imakhlin) [inline pkg/partmgr/partmgr.go]: days or hours?

File: pkg/partmgr/partmgr.go [hunk 7]
```diff
@@ -165,55 +226,47 @@ func (p *DBPartition) Time2Bucket(t int64) int {
...
-	}
-
-	dayIndex := d % p.days
-	chunkIdx := dayIndex*24/p.hoursInChunk + h/p.hoursInChunk
-	return chunkIdx
+	return int((tmilli - p.startTime) / IntervalInMilli(p.manager.cfg.PartitionSchemaInfo.ChunckerInterval)) + 1
 }
 
 // is t covered by this partition
 func (p *DBPartition) InRange(t int64) bool {
```
Comment (imakhlin) [inline pkg/partmgr/partmgr.go]: `InRange` -> `IsInRange`

File: pkg/partmgr/partmgr.go [hunk 8]
```diff
@@ -267,3 +326,22 @@ func TimeToDHM(tmilli int64) (int, int) {
 	d := t / 3600 / 24
 	return d, h
 }
+
+func IntervalInMilli(format string) int64 {
+	interval, _ := strconv.Atoi(format[0 : len(format)-1])
+	unit := format[len(format)-1]
```
Comment (yaronha) [inline pkg/partmgr/partmgr.go]: @dinal need to check len>0

File: pkg/aggregate/aggregate.go [hunk 9]
```diff
@@ -57,11 +58,43 @@ var aggrToString = map[AggrType]string{
...
+	"sqr":    {Name: "_v_sqr", Type: "array", Nullable: true, Items: "double"},
+	"max":    {Name: "_v_max", Type: "array", Nullable: true, Items: "double"},
+	"min":    {Name: "_v_min", Type: "array", Nullable: true, Items: "double"},
+	"last":   {Name: "_v_last", Type: "array", Nullable: true, Items: "double"},
+	"avg":    {Name: "_v_avg", Type: "array", Nullable: true, Items: "double"},
+	"rate":   {Name: "_v_rate", Type: "array", Nullable: true, Items: "double"},
+	"stddev": {Name: "_v_stddev", Type: "array", Nullable: true, Items: "double"},
+	"stdvar": {Name: "_v_stdvar", Type: "array", Nullable: true, Items: "double"},
+}
+
```
Comment (yaronha) [inline pkg/aggregate/aggregate.go]: @dinal `v` is a param (col name), not a fixed value

**Pull request merged**